packages/client/hooks/useActiveTopTemplate.ts

```diff
@@ -29,7 +29,7 @@ const useActiveTopTemplate = (
         })
       }
     }, 300)
-  }, [isActive, selectedTemplateId])
+  }, [isActive, selectedTemplateId, edges])
```

Comment:
This was the bug. The `selectedTemplateId` would initially be the optimistic id, but once it changed to the id from the server, `edges` wouldn't update - it would still contain the optimistic id so `isSelectedInView` would be false